File: backend/app/schemas/mine_transport.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel


class MineTransportBase(BaseModel):
    date_jalali: str
    date_gregorian: date
    truck_id: uuid.UUID
    driver_id: uuid.UUID
    receipt_no: str
    tonnage_kg: int
    destination_id: uuid.UUID
    cost_per_kg: int
    notes: str | None = None


class MineTransportCreate(MineTransportBase):
    pass


class MineTransportUpdate(BaseModel):
    date_jalali: str | None = None
    date_gregorian: date | None = None
    truck_id: uuid.UUID | None = None
    driver_id: uuid.UUID | None = None
    receipt_no: str | None = None
    tonnage_kg: int | None = None
    destination_id: uuid.UUID | None = None
    cost_per_kg: int | None = None
    notes: str | None = None


class MineTransportPaymentUpdate(BaseModel):
    payment_date_jalali: str
    payment_date_gregorian: date


class TruckInfo(BaseModel):
    id: uuid.UUID
    plate_number: str

    model_config = {"from_attributes": True}


class DriverInfo(BaseModel):
    id: uuid.UUID
    full_name: str

    model_config = {"from_attributes": True}


class SiteInfo(BaseModel):
    id: uuid.UUID
    code: str
    name_fa: str
    name_en: str

    model_config = {"from_attributes": True}


class MineTransportResponse(MineTransportBase):
    id: uuid.UUID
    bill_of_lading_image: str | None
    status: str
    payment_date_jalali: str | None
    payment_date_gregorian: date | None
    payment_receipt_image: str | None
    is_paid: bool
    created_at: datetime
    updated_at: datetime
    created_by: uuid.UUID | None
    truck: TruckInfo | None
    driver: DriverInfo | None
    destination: SiteInfo | None

    model_config = {"from_attributes": True}


class MineTransportListResponse(BaseModel):
    items: list[MineTransportResponse]
    total: int
    page: int
    size: int

File: frontend/src/api/mine_transport.ts
import apiClient from './client'

export interface MineTransport {
  id: number
  date: string
  truck_id: number
  driver_id: number
  destination: string
  tonnage_kg: number
  cost_per_kg: number
  receipt_no: string
  bill_of_lading_url?: string
  is_paid: boolean
  payment_date?: string
  payment_receipt_url?: string
  notes?: string
  truck?: { id: number; plate_number: string }
  driver?: { id: number; full_name: string }
  created_at: string
}

export interface MineTransportCreate {
  date: string
  truck_id: number
  driver_id: number
  destination: string
  tonnage_kg: number
  cost_per_kg: number
  receipt_no: string
  notes?: string
}

export interface MineTransportPayment {
  payment_date: string
  notes?: string
}

export interface PaginatedMineTransports {
  items: MineTransport[]
  total: number
  page: number
  size: number
}

export const getMineTransports = (params?: Record<string, unknown>) =>
  apiClient.get<PaginatedMineTransports>('/mine-transport', { params }).then(r => r.data)

export const getMineTransport = (id: number) =>
  apiClient.get<MineTransport>(`/mine-transport/${id}`).then(r => r.data)

export const createMineTransport = (data: MineTransportCreate) =>
  apiClient.post<MineTransport>('/mine-transport', data).then(r => r.data)

export const updateMineTransport = (id: number, data: Partial<MineTransportCreate>) =>
  apiClient.put<MineTransport>(`/mine-transport/${id}`, data).then(r => r.data)

export const payMineTransport = (id: number, data: MineTransportPayment) =>
  apiClient.post<MineTransport>(`/mine-transport/${id}/pay`, data).then(r => r.data)

export const deleteMineTransport = (id: number) =>
  apiClient.delete(`/mine-transport/${id}`)

export const uploadBillOfLading = (id: number, file: File) => {
  const fd = new FormData()
  fd.append('file', file)
  return apiClient.post<MineTransport>(`/mine-transport/${id}/bill-of-lading`, fd).then(r => r.data)
}

export const uploadPaymentReceipt = (id: number, file: File) => {
  const fd = new FormData()
  fd.append('file', file)
  return apiClient.post<MineTransport>(`/mine-transport/${id}/payment-receipt`, fd).then(r => r.data)
}

export const importMineTransports = (rows: Record<string, unknown>[]) =>
  apiClient.post<{ success: number; errors: string[] }>('/mine-transport/import', { rows }).then(r => r.data)
